### DISCUSSION TITLE
YARP behind YARP: Why HTTP/2 error code 'HTTP_1_1_REQUIRED' ?

### DISCUSSION THREAD

**Discussion by Sen-Gupta** [dotnet/yarp#2368] [Q&A] [answered]
All proxies are running in docker container. (Lastest YARP)

We have a scenario, where we have a YARP instance (Let's call **internet-proxy**) facing internet running on https. 

This instance forwards a few requests to another YARP instance (Let's call **internal-proxy**) running internally on http(no ssl here).
http://192.168.1.134:5823/
internal-proxy is running with Http2 as default protocol 
"Kestrel": {
   "EndpointDefaults": {
     "Protocols": "Http2"
   }
 },

An http2 requests from client originates and hits (internet-proxy), since it is it a grpc call, we have configured it to be forwarded as http2 and require exact version.

Connection to internal-proxy fails with following:

external-proxy: (makes call to internal-proxy) 
Proxying to http://192.168.1.134:5823/IEquitiesGrpcService/GetAllEntitiesAsync HTTP/2 RequestVersionExact no-streaming

But fails to perform Handshake with following error:

System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: The request was aborted.
 ---> System.IO.IOException: An HTTP/2 connection could not be established because the server did not complete the HTTP/2 handshake.
 ---> System.Net.Http.HttpProtocolException: **The HTTP/2 server closed the connection. HTTP/2 error code 'HTTP_1_1_REQUIRED'** (0xd).
   at System.Net.Http.Http2Connection.ThrowProtocolError(Http2ProtocolErrorCode errorCode, String message)

Any suggestions?

**Comment by Tratcher** [comment, marked as answer]
> no-streaming

This is suspect, that should say 'streaming' for gRPC calls. Can you collect full debug logs for the server?

**Reply by Sen-Gupta** [reply]
Never mind, it was not an issue with YARP.

It is cloudflare that forces Grpc calls to use GrpcWeb.
